src/mcu.cpp

```diff
@@ -1064,12 +1064,16 @@ int MCU_OpenAudio(void)
     spec.channels = 2;
     spec.callback = audio_callback;
     spec.samples = audio_page_size / 4;
+    
+    const char* audioDevicename = SDL_GetAudioDeviceName(deviceIndex, 0);
```

Comment:
maybe add range check for deviceIndex? (against what SDL_GetNumAudioDevices() returns)

Reply:
OK, I have added error checking and fixed default device selection logic ( 0 does not mean not default, but always denotes 1st device).